### DISCUSSION TITLE
About the sense of memos for getting the value of signals

### DISCUSSION THREAD

**Discussion by JaielZeus** [solidjs/solid#1805] [Q&A] [answered]
I always wrap a signal into a memo to reduce extra calculations as stated in the docs. I am not sure if that makes any sense anymore.

Is my new assumption correct that when jsut doing the ``get()`` anywhere in jsx or for example a seperate thread checking on the value is the same as calling the memoized version of the signal? I'm a bit confused what would be the best practice here.

Here an example (check the comments):
```
const [_get,set] = createSignal(0);
const get = createMemo(_get); // does that make sense?

setInterval(()=>{
  console.log(_get());
  console.log(get());  // is the memoized more efficient here?
  }, 10000
)
return (
  <>
    <div> {_get()} {_get()}
    </div>
    <div> {get()} {get()} {/* is the memoized more efficient here?*/}
    </div>
  </>
)
```

**Comment by atk** [comment]
Have a look at makePersisted to see how to pull the same trick without the memo by wrapping the setter: https://github.com/solidjs-community/solid-primitives/tree/main/packages/storage#how-to-use-it

**Reply by JaielZeus** [reply]
Hey, thanks for pointing me there. I will check it out and see hwo I can incorporate it in my code isntead of my custom solution. I updated my question because I am more concerned about wrapping the ``get`` signal in a memo and if that makes any sense and does things more efficient or not in contrast to jsut using the ``get`` signal directly

**Reply by atk** [reply]
While memoization can improve the performance by avoiding unnecessary reactivity, it's not for free. If you merely want to change the timing of your effects, try to `batch` the set operations or use `createDeferred` instead of `createEffect`.

**Reply by JaielZeus** [reply]
@atk 
Would this make sense to do?
```
const [_get,set] = createSignal(0);
const get = createMemo(_get);
```

**Reply by atk** [reply, marked as answer]
It would certainly work, but it would not improve the performance, except maybe in very few edge cases. Unless performance becomes an actual, measurable issue, this falls under premature optimization and should be strictly avoided. Also, as I already remarked, other means of optimization are readily available.

**Reply by JaielZeus** [reply]
Thank you for your input on this issue. I was already thinking both calls to the raw signal ``_get`` and to the memoized ``get`` would be the same in performance because it is simply an accessor being called in the background. I unfortunately am not familiar with the inner workings on solidJS so just going from the documentation is hard on it's own.
